tests/test_utils.py

```diff
@@ -248,38 +248,76 @@ def test_multipart_parallel_upload(self):
-                                      'bundles/001e784d-fbff-4ee4-aefd-6eb2c8cad783.2018-10-08T201847.591231Z',
-                                      'bundles/001650e3-87a0-4145-9560-430930445071.2018-10-09T002318.558884Z',
-                                      'bundles/00193c56-85f3-4c72-b236-21e6359cb933.2019-02-26T035613.863196Z',
-                                      'bundles/001d3eff-b102-4670-9dad-92c9fb6534bb.2019-02-26T033443.407780Z',
-                                      'bundles/001c3895-59ad-4897-967c-d6a33233f65e.2018-09-13T131642.918470Z']
-
-            def list_v2(self, *args, **kwargs):
-                list_tuples = [(x, None) for x in self.tombstoned_bundle_list]
-                return iter(list_tuples)
-        mock_list_v2.return_value = MockStorageHandler()
+        mock_list_v2.return_value = self.MockStorageHandler()
         resp = enumerate_available_bundles(replica='aws')
-        dead_bundles = [x.split('bundles/')[1].rsplit('.', 1)[0] for x in MockStorageHandler.tombstoned_bundle_list
-                        if x.endswith(TOMBSTONE_SUFFIX)]
         for x in resp['bundles']:
-            self.assertNotIn('.'.join([x['uuid'], x['version']]), dead_bundles)
-            self.assertNotIn(x['uuid'], dead_bundles)
-        self.assertEquals(resp['page_count'], 10)
+            self.assertNotIn('.'.join([x['uuid'], x['version']]), self.MockStorageHandler.dead_bundles)
+
+    @mock.patch("dss.Config.get_blobstore_handle")
+    def test_tombstone_pages(self, mock_list_v2):
+        test_per_page = [{"size": 5,
+                          "last_good_bundle": {"uuid": "00176adc-f15d-4e46-8623-96cd047ca3df",
+                                               "version": "2019-02-26T033558.843528Z"}},
+                         {"size": 2,
+                          "last_good_bundle": {"uuid": "001650e3-87a0-4145-9560-430930445071",
+                                               "version": "2018-10-09T002318.558884Z"}},
+                         {"size": 4,
+                          "last_good_bundle": {"uuid": "00171d5a-f14f-48b3-a467-0b0ac8208c6b",
+                                               "version": "2019-02-26T033414.851898Z"}}]
+
+        mock_list_v2.return_value = self.MockStorageHandler()
+        for tests in test_per_page:
+            test_size = tests['size']
+            last_good_bundle = tests['last_good_bundle']
+            resp = enumerate_available_bundles(replica='aws', per_page=test_size)
+            page_one = resp['bundles']
+            for x in resp['bundles']:
+                self.assertNotIn('.'.join([x['uuid'], x['version']]), self.MockStorageHandler.dead_bundles)
+            self.assertDictEqual(last_good_bundle, resp['bundles'][-1])
+            search_after = resp['search_after']
+            resp = enumerate_available_bundles(replica='aws', per_page=2,
+                                               search_after=search_after)
```

Comment:
`token` doesn't need to be passed in?

Reply:
refactored tests to support this

Reply:
i had to change the page strategy to :

    list 5, and then use paging info to list 5 more
    list 2, and then use paging info to list 2 more
    list 4, and then use paging info to list 4 more

i changed the bundle list to support this, it was due to an enumeration error that would occur in the cloud_blobstore.